<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Curriculum Vitae</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link rel="shortcut icon" type="image/x-icon" href="favicon.ico" />
<link rel="icon" type="image/x-icon" href="favicon.ico" />




<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="styles.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">THOM QUINN</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="about-me.html">ABOUT ME</a>
</li>
<li>
  <a href="curriculum-vitae.html">C.V.</a>
</li>
<li>
  <a href="musings.html">MUSINGS</a>
</li>
<li>
  <a href="fiction.html">POETRY</a>
</li>
<li>
  <a href="prints.html">PHOTOGRAPHY</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Curriculum Vitae</h1>

</div>


<div id="education" class="section level3">
<h3>Education</h3>
</div>
<div id="phd" class="section level2">
<h2>PhD</h2>
<div id="deakin-university-september-2016-to-present" class="section level6">
<h6><strong>Deakin University:</strong> September 2016 to present</h6>
</div>
<div id="currently-pursuing-a-doctor-of-philosophy-with-a-concentration-in-bioinformatics-and-medical-genomics" class="section level6">
<h6>Currently pursuing a Doctor of Philosophy with a concentration in bioinformatics and medical genomics</h6>
</div>
</div>
<div id="md" class="section level2">
<h2>MD</h2>
<div id="suny-upstate-medical-university-august-2011-to-may-2016" class="section level6">
<h6><strong>SUNY Upstate Medical University:</strong> August 2011 to May 2016</h6>
</div>
<div id="graduated-with-a-medicinae-doctor-devoting-credited-hours-to-research-in-bioinformatics-and-medical-genomics" class="section level6">
<h6>Graduated with a Medicinae Doctor, devoting credited hours to research in bioinformatics and medical genomics</h6>
</div>
</div>
<div id="bs" class="section level2">
<h2>BS</h2>
<div id="suny-new-paltz-august-2009-to-may-2011" class="section level6">
<h6><strong>SUNY New Paltz:</strong> August 2009 to May 2011</h6>
</div>
<div id="graduated-summa-cum-laude-obtaining-a-bachelors-degree-in-biology-with-a-concentration-in-biotechnology" class="section level6">
<h6>Graduated <em>summa cum laude</em>, obtaining a Bachelor’s Degree in Biology with a concentration in biotechnology</h6>
</div>
</div>
<div id="as" class="section level2">
<h2>AS</h2>
<div id="suny-orange-january-2007-to-may-2009" class="section level6">
<h6><strong>SUNY Orange:</strong> January 2007 to May 2009</h6>
</div>
<div id="graduated-summa-cum-laude-obtaining-an-associates-degree-in-science-with-a-concentration-in-biology" class="section level6">
<h6>Graduated <em>summa cum laude</em>, obtaining an Associate’s Degree in Science with a concentration in biology</h6>
<p><br></p>
</div>
<div id="open-source-software" class="section level3">
<h3>Open source software</h3>
<div id="r-packages-which-i-designed-and-maintain" class="section level4">
<h4>R packages which I designed and maintain,</h4>
<div id="balance-cran" class="section level5">
<h5><strong>balance</strong> | <a href="https://cran.r-project.org/package=balance">CRAN</a></h5>
<div id="balances-have-become-a-cornerstone-of-compositional-data-analysis.-however-conceptualizing-balances-is-difficult-especially-for-high-dimensional-data.-most-often-investigators-visualize-balances-with-balance-dendrograms.-however-this-visualization-tool-does-not-scale-well-for-large-data.-this-package-provides-an-alternative-scheme-for-visualizing-balances-described-in-quinn-2018.-this-package-also-provides-a-method-for-principal-balance-analysis." class="section level6">
<h6>Balances have become a cornerstone of compositional data analysis. However, conceptualizing balances is difficult, especially for high-dimensional data. Most often, investigators visualize balances with “balance dendrograms”. However, this visualization tool does not scale well for large data. This package provides an alternative scheme for visualizing balances, described in [Quinn (2018)]. This package also provides a method for principal balance analysis.</h6>
</div>
</div>
<div id="exprso-cran" class="section level5">
<h5><strong>exprso</strong> | <a href="https://CRAN.R-project.org/package=exprso">CRAN</a></h5>
<div id="supervised-machine-learning-has-an-increasingly-important-role-in-data-analysis.-this-package-introduces-a-framework-for-rapidly-building-and-deploying-supervised-machine-learning-in-a-high-throughput-manner.-this-package-provides-a-user-friendly-interface-that-empowers-investigators-to-execute-state-of-the-art-binary-and-multi-class-classification-as-well-as-regression-with-minimal-programming-experience-necessary." class="section level6">
<h6>Supervised machine learning has an increasingly important role in data analysis. This package introduces a framework for rapidly building and deploying supervised machine learning in a high-throughput manner. This package provides a user-friendly interface that empowers investigators to execute state-of-the-art binary and multi-class classification, as well as regression, with minimal programming experience necessary.</h6>
</div>
</div>
<div id="peakram-cran" class="section level5">
<h5><strong>peakRAM</strong> | <a href="https://CRAN.R-project.org/package=peakRAM">CRAN</a></h5>
<div id="when-working-with-big-data-sets-ram-conservation-is-critically-important.-however-it-is-not-always-enough-to-just-monitor-the-size-of-the-objects-created.-so-called-copy-on-modify-behavior-characteristic-of-r-means-that-some-expressions-or-functions-may-require-an-unexpectedly-large-amount-of-ram-overhead.-for-example-replacing-a-single-value-in-a-matrix-duplicates-that-matrix-in-the-back-end-making-this-task-require-twice-as-much-ram-as-that-used-by-the-matrix-itself.-this-package-makes-it-easy-to-monitor-the-total-and-peak-ram-used-so-that-developers-can-quickly-identify-and-eliminate-ram-hungry-code." class="section level6">
<h6>When working with big data sets, RAM conservation is critically important. However, it is not always enough to just monitor the size of the objects created. So-called “copy-on-modify” behavior, characteristic of R, means that some expressions or functions may require an unexpectedly large amount of RAM overhead. For example, replacing a single value in a matrix duplicates that matrix in the back-end, making this task require twice as much RAM as that used by the matrix itself. This package makes it easy to monitor the total and peak RAM used so that developers can quickly identify and eliminate RAM hungry code.</h6>
</div>
</div>
<div id="propr-cran" class="section level5">
<h5><strong>propr</strong> | <a href="https://CRAN.R-project.org/package=propr">CRAN</a></h5>
<div id="the-bioinformatic-evaluation-of-gene-co-expression-often-begins-with-correlation-based-analyses.-however-correlation-lacks-validity-when-applied-to-relative-data-including-count-data-generated-by-next-generation-sequencing.-this-package-implements-several-metrics-for-proportionality-including-phi-lovell-et-al-2015-and-rho-erb-and-notredame-2016.-this-package-also-implements-several-metrics-for-differential-proportionality.-unlike-correlation-these-measures-give-the-same-result-for-both-relative-and-absolute-data." class="section level6">
<h6>The bioinformatic evaluation of gene co-expression often begins with correlation-based analyses. However, correlation lacks validity when applied to relative data, including count data generated by next-generation sequencing. This package implements several metrics for proportionality, including phi [Lovell et al (2015)] and rho [Erb and Notredame (2016)]. This package also implements several metrics for differential proportionality. Unlike correlation, these measures give the same result for both relative and absolute data.</h6>
</div>
</div>
</div>
<div id="r-packages-to-which-i-contributed" class="section level4">
<h4>R packages to which I contributed,</h4>
<div id="aldex2-bioconductor" class="section level5">
<h5><strong>ALDEx2</strong> | <a href="https://www.bioconductor.org/packages/devel/bioc/html/ALDEx2.html">Bioconductor</a></h5>
<div id="this-package-performs-a-differential-abundance-analysis-of-biological-count-data-using-log-ratio-transformations." class="section level6">
<h6>This package performs a differential abundance analysis of biological count data using log-ratio transformations.</h6>
</div>
</div>
<div id="catmap-cran" class="section level5">
<h5><strong>catmap</strong> | <a href="https://CRAN.R-project.org/package=catmap">CRAN</a></h5>
<div id="this-package-conducts-a-meta-analysis-on-combined-case-control-and-family-based-tdt-genetic-data." class="section level6">
<h6>This package conducts a meta-analysis on combined case-control and family-based (TDT) genetic data.</h6>
<p><br></p>
</div>
</div>
</div>
</div>
<div id="professional-experience" class="section level3">
<h3>Professional experience</h3>
<div id="in-industry" class="section level4">
<h4>In industry,</h4>
<div id="adveq-management-april-2016-to-august-2017" class="section level5">
<h5><strong>Adveq Management:</strong> April 2016 to August 2017</h5>
<div id="employed-as-a-data-scientist-to-develop-an-object-oriented-back-end-in-r-for-organizing-modeling-and-analyzing-transactions-between-financial-institutions.-position-held-as-a-short-term-free-lance-contract." class="section level6">
<h6>Employed as a <strong>Data Scientist</strong> to develop an object-oriented back-end in R for organizing, modeling, and analyzing transactions between financial institutions. Position held as a short-term, free-lance contract.</h6>
</div>
</div>
</div>
<div id="in-research" class="section level4">
<h4>In research,</h4>
<div id="deakin-university-january-2018-to-present" class="section level5">
<h5><strong>Deakin University:</strong> January 2018 to present</h5>
<div id="employed-as-a-casual-research-fellow-in-the-pattern-recognition-and-data-analysis-prada-lab-under-dr.svetha-venkatesh-to-study-autism-spectrum-disorder.-this-role-involved-pre-processing-genomic-data-building-and-deploying-classifiers-based-on-gene-expression-signatures-and-supervising-a-research-assistant." class="section level6">
<h6>Employed as a <strong>Casual Research Fellow</strong> in the Pattern Recognition and Data Analysis (PRaDA) lab under Dr. Svetha Venkatesh to study autism spectrum disorder. This role involved pre-processing genomic data, building and deploying classifiers based on gene expression signatures, and supervising a research assistant.</h6>
</div>
</div>
<div id="suny-upstate-medical-university-august-2012-to-may-2016" class="section level5">
<h5><strong>SUNY Upstate Medical University:</strong> August 2012 to May 2016</h5>
<div id="employed-as-a-student-research-assistant-in-the-psychiatric-genetic-epidemiology-and-neurobiology-psychgene-lab-under-dr.stephen-glatt-to-study-the-molecular-underpinnings-of-psychiatric-disorders.-this-role-involved-the-development-of-bioinformatic-tools-for-the-analysis-of-single-nucleotide-polymorphisms-and-microarray-based-gene-expression-data.-extensive-time-devoted-to-the-implementation-machine-learning-for-the-classification-of-autism-spectrum-disorder." class="section level6">
<h6>Employed as a <strong>Student Research Assistant</strong> in the Psychiatric Genetic Epidemiology and Neurobiology (PsychGENe) lab under Dr. Stephen Glatt to study the molecular underpinnings of psychiatric disorders. This role involved the development of bioinformatic tools for the analysis of single nucleotide polymorphisms and microarray-based gene expression data. Extensive time devoted to the implementation machine learning for the classification of autism spectrum disorder.</h6>
</div>
</div>
<div id="suny-new-paltz-september-2009-to-august-2010" class="section level5">
<h5><strong>SUNY New Paltz:</strong> September 2009 to August 2010</h5>
<div id="employed-as-a-student-research-assistant-in-an-organic-chemistry-lab-under-dr.frantz-folmer-andersen-to-study-synthetic-diamine-macrocycles-which-have-the-capability-to-distinguish-between-mirror-image-molecules." class="section level6">
<h6>Employed as a <strong>Student Research Assistant</strong> in an organic chemistry lab under Dr. Frantz Folmer-Andersen to study synthetic diamine macrocycles which have the capability to distinguish between mirror image molecules.</h6>
</div>
</div>
<div id="suny-purchase-may-2009-to-july-2009" class="section level5">
<h5><strong>SUNY Purchase:</strong> May 2009 to July 2009</h5>
<div id="employed-as-a-student-research-assistant-in-an-environmental-science-lab-under-dr.kristopher-baker-to-study-the-impact-of-anthropogenic-input-on-the-biodiversity-of-the-hudson-river-estuary." class="section level6">
<h6>Employed as a <strong>Student Research Assistant</strong> in an environmental science lab under Dr. Kristopher Baker to study the impact of anthropogenic input on the biodiversity of the Hudson River estuary.</h6>
</div>
</div>
</div>
<div id="in-education" class="section level4">
<h4>In education,</h4>
<div id="deakin-university-march-2017-to-may-2019" class="section level5">
<h5><strong>Deakin University:</strong> March 2017 to May 2019</h5>
<div id="employed-as-a-graduate-assistant-for-two-courses-titled-principles-of-pharmacology-x3-and-therapeutic-development-x2." class="section level6">
<h6>Employed as a <strong>Graduate Assistant</strong> for two courses titled, <em>Principles of Pharmacology</em> (x3) and <em>Therapeutic Development</em> (x2).</h6>
</div>
</div>
<div id="coursera-september-2016-to-october-2016" class="section level5">
<h5><strong>Coursera:</strong> September 2016 to October 2016</h5>
<div id="employed-as-a-graduate-assistant-for-an-online-course-titled-big-data-genes-and-medicine." class="section level6">
<h6>Employed as a <strong>Graduate Assistant</strong> for an online course titled, <em>Big Data, Genes, and Medicine</em>.</h6>
</div>
</div>
<div id="suny-new-paltz-september-2009-to-may-2011" class="section level5">
<h5><strong>SUNY New Paltz:</strong> September 2009 to May 2011</h5>
<div id="employed-as-a-teaching-assistant-for-an-undergraduate-organic-chemistry-laboratory-class." class="section level6">
<h6>Employed as a <strong>Teaching Assistant</strong> for an undergraduate organic chemistry laboratory class.</h6>
</div>
</div>
<div id="suny-orange-september-2008-to-august-2009" class="section level5">
<h5><strong>SUNY Orange:</strong> September 2008 to August 2009</h5>
<div id="employed-as-a-tutor-in-a-university-math-learning-center." class="section level6">
<h6>Employed as a <strong>Tutor</strong> in a university math learning center.</h6>
<p><br></p>
</div>
</div>
</div>
</div>
<div id="community-engagement" class="section level3">
<h3>Community engagement</h3>
<div id="geelong-hacky-hour-october-2017-to-present" class="section level5">
<h5><strong>Geelong Hacky Hour:</strong> October 2017 to present</h5>
<div id="co-founder-of-the-geelong-hacky-hour-chapter-where-researchers-help-other-researchers-learn-to-program." class="section level6">
<h6>Co-founder of the Geelong Hacky Hour chapter where researchers help other researchers learn to program.</h6>
</div>
</div>
<div id="abacbs-combine-october-2018" class="section level5">
<h5><strong>ABACBS &amp; COMBINE:</strong> October 2018</h5>
<div id="reviewed-abstracts-as-part-of-the-abacbs-combine-2018-program-committee." class="section level6">
<h6>Reviewed abstracts as part of the ABACBS &amp; COMBINE 2018 Program Committee.</h6>
<p><br></p>
</div>
</div>
</div>
<div id="awards-and-scholarships" class="section level3">
<h3>Awards and scholarships</h3>
<p>2017 Travel Award to Winter School in Mathematical &amp; Computational Biology</p>
<p>2016 Travel Award to 2nd Bioconductor Asia-Pacific Developers’ Meeting</p>
<p>2014 Student Research Celebration Presenter Research Project Award</p>
<p>2010 New Paltz Student Undergraduate Research Experience (SURE) stipend</p>
<p>2009 Purchase Bridges to Baccalaureate Program stipend</p>
<p>2009 Outgoing Faculty Annual Scholarship awarded by SUNY Orange</p>
<p>2009 Outgoing William F. Ehlers Jr. Memorial Scholarship for excellence in anthropology awarded by SUNY Orange</p>
<p>2009 Outgoing Dr. Cortland R. Mapes Scholarship for excellence in biology awarded by SUNY Orange</p>
<p>2008 American Chemical Society award for excellence in chemistry</p>
<p>2007-2009 Recipient of President’s Scholarship for Outstanding Academic Achievement at SUNY Orange</p>
<p><br></p>
</div>
<div id="presentations" class="section level3">
<h3>Presentations</h3>
<p><strong>Quinn TP</strong>. June 2019. Using balances to engineer features for the classification of health biomarkers: a new approach to balance selection. Invited talk presented at CoDaWork 2019 in Terrassa, Spain.</p>
<p><strong>Quinn TP</strong>. March 2019. A practical guide to (supervised) machine learning using the exprso R package. Invited talk presented at Deakin University in Geelong, Victoria.</p>
<p><strong>Quinn TP</strong>. October 2018. Understanding sequencing data as compositions: theory and application. Invited talk presented at Centre for Genomic Regulation in Barcelona, Spain.</p>
<p><strong>Quinn TP</strong>. October 2017. A compositionally valid pipeline for any-omics data. Invited abstract presented at Australasian Genomic Technologies Association in Hobart, Tasmania.</p>
<p><br></p>
</div>
<div id="peer-review-contributions" class="section level3">
<h3>Peer-review contributions</h3>
<p>Wellcome Trust</p>
<p>Statistical Methods in Medical Research</p>
<p>PeerJ</p>
<p>American Journal of Medical Genetics B</p>
<p><br></p>
</div>
<div id="publications" class="section level3">
<h3>Publications</h3>
<p><strong>Quinn TP+</strong>, Lee SC+, Venkatesh S, &amp; Nguyen T. 2019. Improving the classification of neuropsychiatric conditions using gene ontology terms as features. American Journal of Medical Genetics B Neuropsychiatric Genetics. Early View. <a href="http://doi.org/10.1002/ajmg.b.32727" class="uri">http://doi.org/10.1002/ajmg.b.32727</a>.</p>
<p><strong>Quinn TP</strong> &amp; Erb I. 2019. Using balances to engineer features for the classification of health biomarkers: a new approach to balance selection. bioRxiv 600122. <a href="http://doi.org/10.1101/600122" class="uri">http://doi.org/10.1101/600122</a>.</p>
<p>Nguyen T, Lee SC, <strong>Quinn TP</strong>, et al., &amp; Le TD. 2019. Personalized Annotation-based Networks (PAN) for the Prediction of Breast Cancer Relapse. bioRxiv 534628. <a href="http://doi.org/10.1101/534628" class="uri">http://doi.org/10.1101/534628</a>.</p>
<p>Beykikhoshk A, <strong>Quinn TP</strong>, Lee SC, Tran T, &amp; Venkatesh S. 2019. DeepTRIAGE: Interpretable and Individualised Biomarker Scores using Attention Mechanism for the Classification of Breast Cancer Sub-types. bioRxiv 533406. <a href="http://doi.org/10.1101/533406" class="uri">http://doi.org/10.1101/533406</a>.</p>
<p>Lee SC+, <strong>Quinn TP+</strong>, et al., &amp; Nguyen T. 2018. Solving for X: evidence for sex-specific autism biomarkers across multiple transcriptomic studies. American Journal of Medical Genetics B Neuropsychiatric Genetics. Early View. <a href="http://doi.org/10.1002/ajmg.b.32701" class="uri">http://doi.org/10.1002/ajmg.b.32701</a>.</p>
<p>Lee SC+, Quinn A, Nguyen T, Venkatesh S, &amp; <strong>Quinn TP+</strong>. 2018. A cross-cancer metastasis signature in the microRNA-mRNA axis of paired tissue samples. bioRxiv 484816. <a href="http://doi.org/10.1101/484816" class="uri">http://doi.org/10.1101/484816</a>.</p>
<p><strong>Quinn TP</strong>, Erb I, Gloor G, Notredame C, Richardson MF, &amp; Crowley TM. 2018. A field guide for the compositional analysis of any-omics data. bioRxiv 484766. <a href="http://doi.org/10.1101/484766" class="uri">http://doi.org/10.1101/484766</a>.</p>
<p><strong>Quinn TP</strong>, Nguyen T, Lee SC, &amp; Venkatesh S. 2018. Cancer as a tissue anomaly: classifying tumor transcriptomes based only on healthy data. bioRxiv 426395. <a href="http://doi.org/10.1101/426395" class="uri">http://doi.org/10.1101/426395</a>.</p>
<p><strong>Quinn TP</strong>. 2018. Visualizing Balances of Compositional Data: A New Alternative to Balance Dendrograms [version 1; referees: 2 approved]. F1000Research 7:1278. <a href="http://doi.org/10.12688/f1000research.15858.1" class="uri">http://doi.org/10.12688/f1000research.15858.1</a>.</p>
<p><strong>Quinn TP</strong>, Crowley TM, &amp; Richardson MF. 2018. Benchmarking differential expression analysis tools for RNA-Seq: normalization-based vs. log-ratio transformation-based methods. BMC Bioinformatics 19(1). <a href="http://doi.org/10.1186/s12859-018-2261-8" class="uri">http://doi.org/10.1186/s12859-018-2261-8</a>.</p>
<p>Allnutt TR, Wade B, <strong>Quinn TP</strong>, Richardson MF, &amp; Crowley TM. 2018. Shortlisting Aptamer Candidates from HT-SELEX data. Aptamers 2. <a href="http://japtamers.co.uk/shortlisting-aptamer-candidates-from-ht-selex-data" class="uri">http://japtamers.co.uk/shortlisting-aptamer-candidates-from-ht-selex-data</a>.</p>
<p><strong>Quinn TP</strong>, Erb I, Richardson MF, &amp; Crowley TM. 2018. Understanding sequencing data as compositions: an outlook and review. Bioinformatics 34(16). <a href="http://doi.org/10.1093/bioinformatics/bty175" class="uri">http://doi.org/10.1093/bioinformatics/bty175</a>.</p>
<p>Erb I, <strong>Quinn T</strong>, Lovell D, &amp; Notredame C. 2017. Differential Proportionality - A Normalization-Free Approach To Differential Gene Expression. Proceedings of CoDaWork 2017, The 7th Compositional Data Analysis Workshop; available under bioRxiv 134536. <a href="http://dx.doi.org/10.1101/134536" class="uri">http://dx.doi.org/10.1101/134536</a>.</p>
<p>Tylee DS, Kikinis Z, <strong>Quinn TP</strong>, et al., &amp; Makris N. 2017. Machine-learning classification of 22q11.2 deletion syndrome: A diffusion tensor imaging study. NeuroImage: Clinical 15(1):832-42. <a href="http://dx.doi.org/10.1016/j.nicl.2017.04.029" class="uri">http://dx.doi.org/10.1016/j.nicl.2017.04.029</a>.</p>
<p><strong>Quinn T</strong>, Richardson MF, Lovell D, &amp; Crowley T. 2017. propr: An R-package for Identifying Proportionally Abundant Features Using Compositional Data Analysis. Scientific Reports 7:16252. <a href="http://dx.doi.org/10.1038/s41598-017-16520-0" class="uri">http://dx.doi.org/10.1038/s41598-017-16520-0</a>.</p>
<p>Englebert C, <strong>Quinn T</strong>, &amp; Bichindaritz I. 2017. Feature selection for survival analysis in bioinformatics. CEUR Workshop Proceedings 1942:30-35. <a href="http://ceur-ws.org/Vol-1942/paper5.pdf" class="uri">http://ceur-ws.org/Vol-1942/paper5.pdf</a>.</p>
<p><strong>Quinn T</strong>, Tylee D, &amp; Glatt S. 2017. exprso: an R-package for the rapid implementation of machine learning algorithms [version 2; referees: 2 approved]. F1000Research 5:2588. <a href="http://dx.doi.org/10.12688/f1000research.9893.2" class="uri">http://dx.doi.org/10.12688/f1000research.9893.2</a>.</p>
<p>Tylee DS, Hess JL, <strong>Quinn TP</strong>, et al., &amp; Glatt SJ. 2016. Blood transcriptomic comparison of individuals with and without autism spectrum disorder: A combined-samples mega-analysis. American Journal of Medical Genetics B Neuropsychiatric Genetics 174(3):181-201. <a href="http://dx.doi.org/10.1002/ajmg.b.32511" class="uri">http://dx.doi.org/10.1002/ajmg.b.32511</a>.</p>
<p>Bichindaritz IB, Cole E, et al., &amp; <strong>Quinn TP</strong>. 2016. Machine Learning Based Automatic Multilevel Stress Detection from ECG Signals. IJCAI 2016: Workshop on Knowledge Discovery in Healthcare Data. <a href="http://sites.google.com/site/ijcai2016kdhealth/accepted-papers" class="uri">http://sites.google.com/site/ijcai2016kdhealth/accepted-papers</a>.</p>
<p>Hess JL, <strong>Quinn TP</strong>, Akbarian S, &amp; Glatt SJ. 2015. Bioinformatic analyses and conceptual synthesis of evidence linking ZNF804A to risk for schizophrenia and bipolar disorder. American Journal of Medical Genetics B Neuropsychiatric Genetics 168(1):14-35. <a href="http://dx.doi.org/10.1002/ajmg.b.32284" class="uri">http://dx.doi.org/10.1002/ajmg.b.32284</a>.</p>
<p><strong>Quinn TP</strong>, Atwood PD, Tanski JM, Moore TF, &amp; Folmer-Andersen J. 2011. Aza-crown macrocycles as chiral solvating agents for mandelic acid derivatives. Journal of Organic Chemistry 76(24):10020-30. <a href="http://dx.doi.org/10.1021/jo2018203" class="uri">http://dx.doi.org/10.1021/jo2018203</a>.</p>
<p><br></p>
</div>
</div>

<div style="position: relative">
  <p style="position: relative; bottom: 0; width:100%; text-align: center; font-size: small">
    &copy; 2019 | <a href="mailto:thom@tpq.me">thom@tpq.me</a> |
    <a href="http://twitter.com/tpq__">Twitter</a> |
    <a href="http://github.com/tpq"> GitHub</a>
  </p>
</div>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
